gossip/api/types.go

```diff
@@ -211,10 +213,24 @@ type GetSourceKeysResponse struct {
 	Entries []*SourceKey `json:"entries"`
 }
 
+// Possible values for SourceKey.Kind
+const (
```

Comment:
Design question: 
It looks like there's a duplicate `Kind` enum in config.proto and, as a result, various bits of code to translate between the two enums. 

1. Is this intentional?   If so, can we find a way to centralize and test the enum conversions so we don't let them get out of sync?
2. Could we pick one of them to be authoritative and drop the other?  Thus we could avoid the above conversion logic / complexity / bug surface area.

Reply:
1) Yes, it's intentional (at least somewhat) -- the external, user-facing API is defined as an HTTP+JSON interface, whereas the internal config file format is text-proto.  The `kind` field is semantically an enum, so is best expressed as an `enum` in the lattter, but JSON doesn't have enums.  Which leads on to&hellip;

2) The other possibility to allow closer synchronization would be to express the enum as an JSON integer rather than a string, and use the same values.  However, that would make the user-facing API a bit less self-explanatory, and there would probably also need to be another set of `const` values to keep in sync anyway.  

(If `configpb.pb.go` has the only/master list of constants, then external clients get put in the position of including a hub-internal file; if `api/types.go` has the only/master list of constants, then the internal config proto file can have `enum` behaviour.)